File: Xjie/Xjie/Models/FeatureFlagModels.swift
import Foundation

// MARK: - Feature Flags (公共 API)

/// GET /api/feature-flags → { flags: { key: enabled } }
struct FeatureFlagClientResponse: Decodable {
    let flags: [String: Bool]
}

// MARK: - Admin Feature Flags

struct AdminFeatureFlag: Codable, Identifiable {
    let id: Int
    let key: String
    var enabled: Bool
    var description: String
    var rollout_pct: Int
    let updated_at: String?
}

struct AdminFeatureFlagList: Decodable {
    let flags: [AdminFeatureFlag]
}

struct AdminFeatureFlagUpdate: Encodable {
    var enabled: Bool?
    var description: String?
    var rollout_pct: Int?
}

struct AdminFeatureFlagCreate: Encodable {
    let key: String
    var enabled: Bool = true
    var description: String = ""
    var rollout_pct: Int = 100
}

// MARK: - Admin Skills

struct AdminSkill: Codable, Identifiable, Hashable {
    let id: Int
    let key: String
    var name: String
    var description: String
    var enabled: Bool
    var priority: Int
    var trigger_hint: String
    var prompt_template: String
    let updated_at: String?

    static func == (lhs: AdminSkill, rhs: AdminSkill) -> Bool { lhs.id == rhs.id }
    func hash(into hasher: inout Hasher) { hasher.combine(id) }
}

struct AdminSkillList: Decodable {
    let skills: [AdminSkill]
}

struct AdminSkillUpdate: Encodable {
    var name: String?
    var description: String?
    var enabled: Bool?
    var priority: Int?
    var trigger_hint: String?
    var prompt_template: String?
}

struct AdminSkillCreate: Encodable {
    let key: String
    let name: String
    var description: String = ""
    var enabled: Bool = true
    var priority: Int = 100
    var trigger_hint: String = ""
    var prompt_template: String = ""
}

File: Android/app/src/main/java/com/xjie/app/core/model/FeatureFlagModels.kt
package com.xjie.app.core.model

import kotlinx.serialization.Serializable

@Serializable
data class FeatureFlagClientResponse(val flags: Map<String, Boolean>)

@Serializable
data class AdminFeatureFlag(
    val id: Int,
    val key: String,
    val enabled: Boolean,
    val description: String,
    val rollout_pct: Int,
    val updated_at: String? = null,
)

@Serializable
data class AdminFeatureFlagList(val flags: List<AdminFeatureFlag>)

@Serializable
data class AdminFeatureFlagUpdate(
    val enabled: Boolean? = null,
    val description: String? = null,
    val rollout_pct: Int? = null,
)

@Serializable
data class AdminFeatureFlagCreate(
    val key: String,
    val enabled: Boolean = true,
    val description: String = "",
    val rollout_pct: Int = 100,
)

@Serializable
data class AdminSkill(
    val id: Int,
    val key: String,
    val name: String,
    val description: String,
    val enabled: Boolean,
    val priority: Int,
    val trigger_hint: String,
    val prompt_template: String,
    val updated_at: String? = null,
)

@Serializable
data class AdminSkillList(val skills: List<AdminSkill>)

@Serializable
data class AdminSkillUpdate(
    val name: String? = null,
    val description: String? = null,
    val enabled: Boolean? = null,
    val priority: Int? = null,
    val trigger_hint: String? = null,
    val prompt_template: String? = null,
)

@Serializable
data class AdminSkillCreate(
    val key: String,
    val name: String,
    val description: String = "",
    val enabled: Boolean = true,
    val priority: Int = 100,
    val trigger_hint: String = "",
    val prompt_template: String = "",
)
